### ISSUE TITLE
No Backward Gradients During Training.

### ISSUE THREAD

**Issue by yuexihang** [ydtydr/MCTensor#1]
Hello, thank you for your great work!

I noticed that when testing the demos in the "applications" directory, the loss for MCT-Models is significantly higher than that of the Torch-Models. The output losses in `linear_regression.ipynb` are as follows:
```
100%|██████████| 2000/2000 [00:12<00:00, 155.67it/s]
100%|██████████| 2000/2000 [00:17<00:00, 115.48it/s]
100%|██████████| 2000/2000 [00:20<00:00, 97.32it/s] 
final losses for MCT: 67.90413389205933 67.9072804831792 67.90728061159759

100%|██████████| 2000/2000 [00:03<00:00, 543.39it/s]
100%|██████████| 2000/2000 [00:03<00:00, 508.84it/s]
100%|██████████| 2000/2000 [00:03<00:00, 525.16it/s]
final losses for torch: 0.0005063056945800781 7.044320682325635e-12 2.607678329821207e-29
```

Upon debugging, I discovered that the gradients for parameters in the MCT-Models are all `None` after calling loss.backward(), even though I have explicitly set model.train().

My environment details are as follows:
- Python version: 3.8.19
- PyTorch version: 1.12.1    (I also tried Pytorch 1.10.0, same problem observed.)

Looking forward to your insights on this issue.
